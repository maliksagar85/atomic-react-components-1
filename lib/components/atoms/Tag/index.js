export { default } from './Tag';
export { TagVanilla } from './Tag';
