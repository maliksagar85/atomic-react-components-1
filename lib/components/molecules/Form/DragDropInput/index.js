export { default } from './DragDropInput';
export { DDFieldInputVanilla } from './DragDropInput';
