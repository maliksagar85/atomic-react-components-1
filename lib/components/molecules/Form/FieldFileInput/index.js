export { default } from './FieldFileInput';
export { FieldFileInputVanilla } from './FieldFileInput';
