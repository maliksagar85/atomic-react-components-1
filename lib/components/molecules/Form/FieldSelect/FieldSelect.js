import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Label from '../../../atoms/Label';
import Select from '../../../atoms/Select';
import FieldError from '../../../atoms/FieldError';
import styles from './FieldSelect.style';

const FieldSelect = ({
  id,
  className,
  name,
  label,
  onChange,
  labelProps,
  fieldProps,
  register,
  errors,
  ...others
}) => (
  <div className={className}>
    <Label htmlFor={id} {...labelProps}>
      {label}
    </Label>
    <Select {...others} ref={register} name={name} />
    {errors[name] && <FieldError>{errors[name].message}</FieldError>}
  </div>
);

FieldSelect.propTypes = {
  id: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.node.isRequired,
  labelProps: PropTypes.objectOf(PropTypes.object),
  fieldProps: PropTypes.objectOf(PropTypes.object),
  onChange: PropTypes.func,
  register: PropTypes.func.isRequired,
  errors: PropTypes.objectOf(PropTypes.object),
};

FieldSelect.defaultProps = {
  labelProps: {},
  fieldProps: {},
  onChange: null,
  errors: {},
};

const StyledFieldSelect = styled(FieldSelect)`
  ${styles};
`;

export default StyledFieldSelect;

export { FieldSelect as FieldSelectVanilla };
