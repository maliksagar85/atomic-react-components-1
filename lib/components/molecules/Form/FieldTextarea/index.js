export { default, FieldTextareaVanilla } from './FieldTextarea';
