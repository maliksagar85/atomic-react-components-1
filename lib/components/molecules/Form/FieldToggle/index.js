export { default } from './FieldToggle';
export { FieldToggleVanilla } from './FieldToggle';
