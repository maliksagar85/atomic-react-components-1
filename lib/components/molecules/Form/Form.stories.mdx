import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import {
  userFormProps,
  userFormSelectProps,
  fileMultipleInputProps,
  fileSingleInputProps,
  fileDDInputProps,
  userFormNameInputProps,
  userFormEmailProps,
  userFormCommentProps,
  userFormAgeGroupProps,
  userFormTnCProps,
  userFormToggleProps,
} from './Form.mock';
import Form from './Form'
import Button from '../../atoms/Button';

<Meta title="Molecules/Form" component={Form} />

### Default

<Preview>
  <Story name="Default">
    <Form {...userFormProps}>
      <Form.Select {...userFormSelectProps} />
      <Form.Input {...userFormNameInputProps} />
      <Form.Input {...userFormEmailProps} />
      <Form.RadioGroup {...userFormAgeGroupProps} />
      <Form.Checkbox {...userFormTnCProps} />
      <Form.Textarea {...userFormCommentProps} />
      <Form.Toggle {...userFormToggleProps} />
      <Form.FieldFile {...fileSingleInputProps} />
      <Form.FieldFile {...fileMultipleInputProps} />
      <Form.DDFile {...fileDDInputProps} />
      <Button primary type="submit">
        Submit
      </Button>
    </Form>
  </Story>
</Preview>

## Props ðŸ”§

<Props of={Form} />


