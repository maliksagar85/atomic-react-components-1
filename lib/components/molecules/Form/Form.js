import React from 'react';
import { useForm } from 'react-hook-form';
import PropTypes from 'prop-types';

import { buildSchema } from '../../../core/utils/buildValidationSchema';
import Debug from './Debug';
import FieldInput from './FieldInput';
import FieldSelect from './FieldSelect';
import FieldTextarea from './FieldTextarea';
import FieldRadioGroup from './FieldRadioGroup';
import FieldCheckbox from './FieldCheckbox';
import FieldToggle from './FieldToggle';
import FieldFileInput from './FieldFileInput';
import DDFileInput from './DragDropInput';

const Form = ({
  onSubmit,
  initialValues,
  children,
  debug,
  validate,
  validationSchema,
  yupValidationSchema,
}) => {
  const getSchema = () => {
    if (yupValidationSchema) return yupValidationSchema;
    return validationSchema ? buildSchema(validationSchema) : null;
  };
  const { handleSubmit, register, errors, setValue } = useForm({
    mode: validate,
    validationSchema: getSchema(yupValidationSchema, validationSchema),
    defaultValues: initialValues,
  });

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      {React.Children.map(children, (child) =>
        React.cloneElement(child, { register, errors, setValue })
      )}
      {debug && <Debug />}
    </form>
  );
};

Form.Checkbox = FieldCheckbox;

Form.Input = FieldInput;

Form.RadioGroup = FieldRadioGroup;

Form.Select = FieldSelect;

Form.Textarea = FieldTextarea;

Form.Toggle = FieldToggle;

Form.FieldFile = FieldFileInput;

Form.DDFile = DDFileInput;

Form.defaultProps = {
  debug: false,
  initialValues: null,
  validationSchema: null,
  yupValidationSchema: null,
  validate: 'onSubmit',
};

Form.propTypes = {
  initialValues: PropTypes.objectOf(PropTypes.object),
  children: PropTypes.node.isRequired,
  onSubmit: PropTypes.func.isRequired,
  debug: PropTypes.bool,
  validationSchema: PropTypes.objectOf(PropTypes.object),
  validate: PropTypes.string,
  yupValidationSchema: PropTypes.func,
};

export default Form;
