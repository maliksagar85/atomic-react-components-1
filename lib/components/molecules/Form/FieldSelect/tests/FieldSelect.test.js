import React from 'react';
import { shallowWithTheme, renderWithTheme } from '../../../../../../__mock__/themeMock';
import { FieldSelectVanilla } from '../FieldSelect';
import Form from '../../Form';
import { ButtonVanilla } from '../../../../atoms/Button';

describe('FieldInput Component', () => {
  let FieldInput = '';

  test('should render correctly', () => {
    FieldInput = shallowWithTheme(
      <Form>
        <div>
          <FieldSelectVanilla />
          <FieldSelectVanilla />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    expect(FieldInput).toMatchSnapshot();
  });

  test('should render label correctly', () => {
    const props = {
      label: 'Select Fav Color',
      options: ['red', 'black', 'blue'],
      name: 'color',
    };
    FieldInput = renderWithTheme(
      <Form>
        <div>
          <FieldSelectVanilla {...props} />
          <FieldSelectVanilla {...props} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    expect(FieldInput.find('label').text()).toEqual('Select Fav ColorSelect Fav Color');
  });

  test('should render select tag', () => {
    const props = {
      label: 'Select Fav Color',
      options: ['red', 'black', 'blue'],
      name: 'color',
    };
    FieldInput = renderWithTheme(
      <Form>
        <div>
          <FieldSelectVanilla {...props} />
          <FieldSelectVanilla {...props} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    expect(FieldInput.find('select')).toHaveLength(2);
  });
});
