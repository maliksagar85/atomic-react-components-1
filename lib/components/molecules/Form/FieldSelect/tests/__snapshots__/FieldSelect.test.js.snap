// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldInput Component should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    errors={Object {}}
    key=".0"
    register={[Function]}
    setValue={[Function]}
  >
    <FieldSelect
      errors={Object {}}
      fieldProps={Object {}}
      labelProps={Object {}}
      onChange={null}
    />
    <FieldSelect
      errors={Object {}}
      fieldProps={Object {}}
      labelProps={Object {}}
      onChange={null}
    />
  </div>
  <Button
    ariaLabel=""
    className=""
    disabled={false}
    elementLocator=""
    errors={Object {}}
    inheritedStyles=""
    key=".1"
    primary={true}
    register={[Function]}
    secondary={false}
    setValue={[Function]}
    tertiary={false}
    transparent={false}
    type="submit"
    variation=""
  >
    Submit
  </Button>
</form>
`;
